app/Jobs/Notifications/CommentNew.php

```diff
@@ -5,46 +5,21 @@
 
 namespace App\Jobs\Notifications;
 
-use App\Exceptions\InvalidNotificationException;
-use App\Models\Comment;
 use App\Models\Follow;
-use App\Models\User;
+use App\Models\Notification;
 
-class CommentNew extends BroadcastNotificationBase
+class CommentNew extends CommentBase
 {
-    protected $comment;
-
-    public function __construct(Comment $comment, User $source)
+    public function getListeningUserIds(): array
     {
-        parent::__construct($source);
-
-        $this->comment = $comment;
+        $userIds = Follow::whereNotifiable($this->comment->commentable)
+            ->where(['subtype' => 'comment']);
 
-        if ($this->comment->commentable === null) {
-            throw new InvalidNotificationException("comment_new: comment #{$this->comment->getKey()} missing commentable");
+        if ($this->comment->parent_id !== null) {
+            $userIds->where('user_id', '<>', $this->comment->parent->user_id);
+            broadcast_notification(Notification::COMMENT_REPLY, $this->comment, $this->source);
```

Comment:
I actually skipped fixing this at the time, because, if you were watching the comment thread but chose to receive only mail notifications for replies, what should show up in your push notifications?